com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -1354,5 +1401,15 @@ void DecideAction()
             m_Info.CopyActions(actions);
             m_ActuatorManager.UpdateActions(actions);
         }
+
+        public void SetTeamManager(ITeamManager teamManager)
```

Comment:
Is the user expected to call SetTeamManager directly or is this hidden from the user?

Reply:
User is expected to call SetTeamManager() to associate an agent with a manager.

Reply:
This has changed after removing cyclic reference between agent and manager. It's now internal and called by TeamManager.RegisterAgent()